flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -2285,7 +2285,19 @@ void OmpStructureChecker::Enter(const parser::OmpClause::Reduction &x) {
   if (CheckReductionOperators(x)) {
     CheckReductionTypeList(x);
   }
+
+  SymbolSourceMap symbols;
+  const auto &ompObjectList{std::get<parser::OmpObjectList>(x.v.t)};
+  GetSymbolsInObjectList(ompObjectList, symbols);
+  for (auto &[symbol, source] : symbols) {
+    if (IsProcedurePointer(*symbol)) {
+      context_.Say(source,
+          "A procedure pointer '%s' must not appear in a REDUCTION clause."_err_en_US,
+          symbol->name());
+    }
+  }
```

Comment:
nit - I am probably being too pedantic here but perhaps this should be in `CheckReductionTypeList`?

Reply:
Will do that before I submit.